wordpress-mobile/WordPress-Android
# Gutenberg - Hide undo/redo buttons in HTML Mode

This PR fixes an issue where the undo/redo buttons in the editor are visible when toggling to HTML mode.

These buttons don't have any functionality while editing in HTML so we should hide them and show them back if the user toggles back to the Visual mode.

To test:

- Open the app
- Open the editor
- Toggle to HTML mode by tapping on the more options (...) three dots in the right area of the navigation bar
- **Expect** to see the HTML editor and the undo/redo buttons to be hidden
- Toggle back to the Visual mode by tapping on the more options (...) three dots in the right area of the navigation bar
- **Expect** to see the undo/redo buttons in the navigation bar

Functionality
-|
<video src="https://github.com/wordpress-mobile/WordPress-Android/assets/4885740/0bed0ac4-8ea0-4971-ba00-41b4334e74dd" width=100 />|

## Regression Notes
1. Potential unintended areas of impact
It affects only the block editor

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and added UI tests

3. What automated tests I added (or what prevented me from doing so)
I added extra steps to `testUndoRedo` for the EditorGutenbergTests, which includes adding a new HTMLEditorScreen.

PR submission checklist:


UI Changes testing checklist:
